TinyInsights.TestApp/MainPage.xaml.cs

```diff
@@ -1,19 +1,44 @@
-﻿namespace TinyInsights.TestApp;
+﻿using Microsoft.Extensions.Logging;
+
+namespace TinyInsights.TestApp;
 
 public partial class MainPage : ContentPage
 {
     private readonly IInsights insights;
     private readonly InsightsMessageHandler insightsMessageHandler;
+    private readonly ILogger logger;
 
-    public MainPage(IInsights insights, InsightsMessageHandler insightsMessageHandler)
+    public MainPage(IInsights insights, InsightsMessageHandler insightsMessageHandler, ILogger logger)
     {
         this.insights = insights;
         this.insightsMessageHandler = insightsMessageHandler;
+        this.logger = logger;
+
+        BindingContext = this;
+
+
         InitializeComponent();
     }
 
+private bool useILogger;
+public bool UseILogger 
+{
+    get => useILogger;
+    set 
+    {
+        useILogger = value;
+        OnPropertyChanged(nameof(UseILogger));
+    }
+}
+
     private async void PageViewButton_OnClicked(object? sender, EventArgs e)
     {
+        if(UseILogger)
+        {
+            logger.LogTrace("MainView");
+            return;
+        }
```

Comment:
are these checks really necessary ? 
Wouldnt be better to implement one signle check into `InsightsExtension.cs` ?

Reply:
No it is just for test purpose. So I can test both ways to log in an easy way.